src/viam/sdk/rpc/dial.hpp

```diff
@@ -12,72 +12,69 @@ namespace sdk {
 class DialOptions;
 class ViamChannel {
    public:
-    std::shared_ptr<grpc::Channel> channel;
     void close();
     ViamChannel(std::shared_ptr<grpc::Channel> channel, const char* path, void* runtime);
     static std::shared_ptr<ViamChannel> dial(const char* uri, boost::optional<DialOptions> options);
 
+    const std::shared_ptr<grpc::Channel>& channel() const;
+
    private:
-    const char* path;
-    bool closed;
-    void* rust_runtime;
+    std::shared_ptr<grpc::Channel> channel_;
+    const char* path_;
+    bool closed_;
+    void* rust_runtime_;
 };
 
 class Credentials {
    public:
-    std::string type_;
-    std::string payload;
-
     Credentials(std::string payload);
+    const std::string& payload() const;
+
+   private:
+    std::string type_;
+    std::string payload_;
 };
 
 class DialOptions {
    public:
-    /// Bypass webRTC and connect directly to the robot
-    // TODO (RSDK-917): This field is currently just for show, we
-    // should update dial logic to actually care about this
-    bool disable_webrtc;
+    DialOptions()
+        : auth_entity_(boost::none), credentials_(boost::none), allow_insecure_downgrade_(false) {}
+
+    const boost::optional<Credentials>& credentials() const;
+    void set_allow_insecure_downgrade(bool allow);
```

Comment:
A fun trick with options types like this is to make the setters return "ref to self":

```
class FooOptions {
public:
    FooOptions& set_x(int x);
    FooOptions& set_y(int y);
};
```

This allows a caller to chain:

```
FooOptions fo;
fo.set_x(42).set_y(4.2);
```

Some people like it, some don't. Not suggesting a change here, but should these options become more complex it can be a nice thing to do for users.